### ISSUE TITLE
nushell 0.84.0 can not import .nu file in config.nu

### ISSUE THREAD

**Issue by Salted-Fish-Swimming** [nushell/nushell#10250]
### Question

nushell's config.nu can't import external files through source/use. If I did, it would report an error after loading in the terminal window.

![图片](https://github.com/nushell/nushell/assets/62016442/d0f3d56d-c898-4139-9266-71b7cf77360a)

### Additional context and details

I've been using nushell for about half a year now, and I recently found a bug while using nushell to manipulate the clipboard through powershell, so I tried to upgrade to the latest version. But I immediately noticed that the new version of nushell does not seem to support the use of source/use in config.nu to import external nu code, is this a feature added in a past update? Or an unresolved bug?

**Comment by fdncred** [comment]
I wonder if this is because we allowed the only NU_LIB_DIRS folder to be commented out in the env.nu?

**Comment by amtoine** [comment]
@Salted-Fish-Swimming 

if `command.nu` is just next to `config.nu`, you can do
```nushell
source ($nu.default-config-dir | path join "command.nu")
```
:yum:

**Comment by Salted-Fish-Swimming** [comment]
> Where is your `command.nu`? It needs to be in some discoverable location, such as in one of NU_LIB_DIRS directories, or right next to `config.nu`. Did it work in 0.83 and stopped working in 0.84?

Yes , `command.nu` is near by `config.nu` , they are in the same `$nu.default-config-dir` path . I'll try `NU_LIB_DIRS` 's env configuration.

**Comment by Salted-Fish-Swimming** [comment]
> I wonder if this is because we allowed the only NU_LIB_DIRS folder to be commented out in the env.nu?

Thanks ! It works ! What's the feature of which version?

**Comment by Salted-Fish-Swimming** [comment]
> @Salted-Fish-Swimming
> 
> if `command.nu` is just next to `config.nu`, you can do
> 
> ```nushell
> source ($nu.default-config-dir | path join "command.nu")
> ```
> 
> 😋

If I set the `$env.NU_LIB_DIRS` it will work ; if i don't set it , it won't work . Will this feature become stable ?

**Comment by fdncred** [comment]
I submitted a PR yesterday where `$env.NU_LIB_DIRS` will be set by default again in our next release.

**Comment by amtoine** [comment]
@fdncred 
i do not think https://github.com/nushell/nushell/pull/10252 solves the issue the OP is having :thinking: 

in https://github.com/nushell/nushell/pull/10252/files, we see that `NU_LIB_DIRS` is set to `~/.config/nushell/scripts/` so a script in `~/.config/nushell/` won't be seen by source by default :open_mouth:

**Comment by Salted-Fish-Swimming** [comment]
> I submitted a PR yesterday where `$env.NU_LIB_DIRS` will be set by default again in our next release.

I strongly recommend against using values other than `$nu.default-config-dir` ( such as `$nu.default-config-dir | path join scripts` ) as default values for bx, at least not until `nushell` can load `command.nu` without `source command.nu`.

Because nushell's vscode plugin can not recognize the need to `source` files from the `scripts` directory through `env.nu`, this can cause the plugin to report a error `File not found`.

**Comment by fdncred** [comment]
> Because nushell's vscode plugin can not recognize the need to source files from the scripts directory through env.nu, this can cause the plugin to report a error File not found.

I'm not quite sure what this means. If you're getting `file not found` with the vscode extension, you probably need to update the extension settings `Include Dirs`.

**Comment by amtoine** [comment]
> > I submitted a PR yesterday where `$env.NU_LIB_DIRS` will be set by default again in our next release.
> 
> I strongly recommend against using values other than `$nu.default-config-dir` ( such as `$nu.default-config-dir | path join scripts` ) as default values for bx
>
do you mean the default `env.nu` should have
```nushell
$env.NU_LIB_DIRS = [
    $nu.default-config-dir
]
```
instead of
```nushell
$env.NU_LIB_DIRS = [
    ($nu.default-config-dir | path join "scripts")
]
```
?

**Comment by fdncred** [comment]
I think we could have both of those paths in NU_LIB_DIRS, but the vscode extension needs to have those defined inside of its settings too.

**Comment by amtoine** [comment]
@Salted-Fish-Swimming 
do you feel like adding `$nu.default-config-dir` to the default `NU_LIB_DIRS`? :innocent:

**Comment by Salted-Fish-Swimming** [comment]
> @Salted-Fish-Swimming do you feel like adding `$nu.default-config-dir` to the default `NU_LIB_DIRS`? 😇

I think a good design is that when you add `$nu.default-config-dir` to `NU_LIB_DIRS` , `config.nu` can no longer `source` other file.

But your current plan is also good, can solve the current problem.

**Comment by Salted-Fish-Swimming** [comment]
> I think we could have both of those paths in NU_LIB_DIRS, but the vscode extension needs to have those defined inside of its settings too.

Yes, I think using `$nu.default-config-dir | path join script` as the default `NU_LIB_DIRS` iff `$nu.default-config-dir | path join script` as the default configuration in vscode extension.

**Comment by fdncred** [comment]
> iff $nu.default-config-dir | path join script as the default configuration in vscode extension.

There is no default configuration for the vscode extension. Feel free to PR such a thing if you feel so inclined.

**Comment by dougg0k** [comment]
Just to add, I was building nushell dotfiles and had similar issues, the solution was making the nushell language server from vscode to recognize the folder. 

```
"nushellLanguageServer.includeDirs": [
	"~/.dotfiles/nushell"
]
```

**Comment by fdncred** [comment]
Yes, include dirs is one of the options in the vscode nushell extension. iirc, it populates NU_LIB_DIRS and helps make finding things easier for the extension. I have 5 folder in it on Windows.
![image](https://github.com/nushell/nushell/assets/343840/823979ff-c6d8-40af-a237-068b707a0e4b)

**Comment by texastoland** [comment]
Is this not nushell/nushell#7849 again? I want that bad enough to check soon!

**Comment by WindSoilder** [comment]
Closing the issue because the question is answered.
